meteor_packages/mats-common/imports/startup/server/data_diff_util.js

```diff
@@ -157,6 +169,12 @@ const getDataForDiffCurve = function (dataset, diffFrom, appParams, isCTC) {
         var tempSubFaArray;
         var tempSubMissArray;
         var tempSubCnArray;
+        var tempSubSquareDiffSumArray;
+        var tempSubNSumArray;
+        var tempSubObsModelDiffSumArray;
+        var tempSubModelSumArray;
+        var tempSubObsSumArray;
+        var tempSubAbsSumArray;
```

Comment:
Do we want these variables available to the entire function? It looks like they're inside a for loop so making them available outside of the loop might result in some hard-to-diagnose bugs.

Generally, it's good practice to start out declaring variables as [`const`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const), switching to [`let`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let) if they need to be modified later, and only using [`var`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var) in special circumstances.

I'd echo this comment for other `var` declarations below.

@randytpierce - what do you think?

Reply:
Well, they definitely can't be `const`, as they are modified by each loop iteration, and Meteor doesn't really use `let`, with `var` assuming that role instead, so I don't think we can change that. I can move them farther down in the code hierarchy, but I don't think it really matters, as there are no other variables in the function with those names?

Reply:
I should've called out that I'd like to see `let` used instead of `var` here. I agree - `const` isn't appropriate if the values are changing.

That being said, I'd be pretty surprised if Meteor was getting in the way of using `let`. `let` was introduced to the Javascript standard in ES6 back in 2015 when `const` was introduced so it's well supported by modern browsers and by Node which Meteor is using. 

Using `var` opens the code up to lots of nasty situations like variable shadowing, re-declaration, hoisting, (the variable can be available before it's been declared) and issues when using closures. If you're curious, I can get into more details on those.

Reply:
I think @randytpierce knows a lot more about this than me, and was the one to suggest using `var` instead of `let` when working with Meteor. Randy, can you elaborate?